### DISCUSSION TITLE
Track reading progress

### DISCUSSION THREAD

**Discussion by Toasterbirb** [mateusz-bak/openreads#698] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'd like to save the current page I'm at into the app. A virtual bookmark you could say. This might also open the doors for more detailed statistics if someone is into that

**Describe the solution you'd like**
- open a book you are reading
- click some sort of button to open a menu that asks how many pages you have read or the new page number you are at
- save the value and show it in the booklist or in the book details

**Describe alternatives you've considered**
A physical bookmark in the real world, but I might have to return the book to library and thus have to remove the bookmark

**Additional context**
This idea mainly comes from the app called [Dante](https://github.com/shockbytes/Dante) that is similar to this app

**Comment by mateusz-bak** [comment]
Hello @Toasterbirb thank you for the submission. I think it would be possible down the road to implement this alongside tracking daily progress of reading. I will keep this in mind.

**Comment by BravishkaSkytano** [comment]
A percentage would be nice too, like you can input the page number or the pecentage. This is great for eBook readers, because the page numbers can change according to the font size, screen size, margins, etc. that we set on our device.

**Comment by Kreuger** [comment]
@mateusz-bak Have you made any progress with this?

**Comment by mateusz-bak** [comment]
Work not started yet.

**Comment by Kreuger** [comment]
Alright thanks.

**Comment by TheAkashicTraveller** [comment]
I would love to see this, I've been looking for something that does this for a while.

**Comment by ikanakova** [comment]
It would be great if it could be manually recorded not only for pages, but also for e.g. chapters, so it could be used for ebooks and short stories on blogs...

**Comment by alensiljak** [comment]
Some texts are tracked by page numbers. Some by percentage.

Some don't support either of these and I like to write down the chapter or section name in those cases.
So, arbitrary text field would be a good option to have in addition to the other two numerical values.

**Comment by jpsnyder** [comment]
I was looking to use this app to replace the [Luna](https://www.lunareadingtracker.com/) app I have been using, because that app seems to crash a lot and is overall buggy. However, the one thing I liked a lot in that app was the tracker feature where you start a timer when you begin reading and then stop it when you are finished. Then it will ask what page you are on. This feature both helps motivate me to read more, keeps my place in a book without needing a bookmark, and gives me an estimated time on how long it will take me to finish the book. It is simple and effective. It's very nice.

I'm hoping to see something like this in OpenReads in the future.

**Comment by tamadamas** [comment]
Time tracking +

I've been trying several apps that were not good enough for me until found yours one.

I've been using this app for a week http://bookbreeze.co for reading a book. 

It has an amazing UI and time reading feature along with stats
![image](https://github.com/mateusz-bak/openreads/assets/1058433/02fdabae-0836-425f-b9e7-1faeac43fc43)
![image](https://github.com/mateusz-bak/openreads/assets/1058433/3b5e0312-2cc4-432f-a945-e218125c1a52)
![image](https://github.com/mateusz-bak/openreads/assets/1058433/1437206f-3b9d-40a2-aedc-4923c6566982)

But it lacks a lot of features that this app offers.
And only support for iOS and no any export options made me looking for other apps.

So, it would be great for time tracking feature to be implemented here as I found this app very promising and good looking for my book journey.

UPDATE: @mateusz-bak I would like to implement this tracking feature if you don't mind.

**Comment by mateusz-bak** [comment]
Definitely @rxx go ahead! After you have something to show we can discuss the implementatio, good luck!

**Comment by 0-BlackSpectrum-0** [comment]
Any updates on this feature?

**Comment by nictakieg0** [comment]
How's it going? Would love to use this feature, as Dante is not receiving any updates already.

**Comment by yashovardhan99** [comment]
Any update on this feature? I see there are no updates for the last 6 months.

**Comment by d-stolarski** [comment]
Any chance for such feature so far? This function is in my opinion very important. I will be happy to stay with the app, but lack of it is unfortunately too big disadventage. 

Thanks a lot in advance for any feedback.

**Comment by mateusz-bak** [comment]
Hello! Yes, I'm planning to release this feature - I just need to refactor how readings are stored to have progress stored.
